hip_data_tools/etl/google_sheet_to_athena.py

```diff
@@ -1,242 +1,104 @@
-            table_name=self.settings.table_name)
-        values = ""
-        if self.settings.partition_value:
-            partition_value_statement = ", '{}'".format(self.settings.partition_value)
-            types += 'STRING'
-        else:
-            partition_value_statement = ''
-        for value in values_matrix:
-            values += "({}{}), ".format(', '.join(self.__get_the_values_list(types, value)),
-                                        partition_value_statement)
-        values = values[:-2]
-        insert_query += values
-        return insert_query
-
-    @staticmethod
-    def __get_the_values_list(types, value):
-        # Return a list of values (Quotes are added if the value type is string or date)
-        # eg: ['2020-03-02', 435, 54, 'customer_1', 34]
-        return ["'{}'".format(
-            val) if data_type.upper() == 'STRING' or data_type.upper() == 'DATE' else
-                "{}".format(
-                    val) for
-                val, data_type
-                in zip(value, types)]
+        return AthenaUtil(database=self.settings.target_database,
+                          conn=AwsConnectionManager(
+                              settings=self.settings.target_connection_settings),
+                          output_bucket=self.settings.target_s3_bucket)
 
-    def load_sheet_to_athena(self, overwrite_table=False):
+    def load_sheet_to_athena(self):
         """
-        Method to load google sheet to athena
-        Args:
-            overwrite_table (boolean): if this is true, it drops the existing athena table and
-                clear the s3 location
+        Load google sheet into Athena
         :return: None
         """
-        sheet_util = self._get_sheets_util()
-        athena_util = self._get_athena_util()
-        s3_util = self._get_s3_util()
-
-        if overwrite_table:
-            athena_util.drop_table(self.settings.table_name)
-            s3_util.delete_recursive(self.settings.s3_dir)
-
-        field_names = []
-        field_types = []
-        if self.settings.fields is None:
-            field_names = sheet_util.get_fields_names(self.settings.workbook_name,
-                                                      self.settings.sheet_name)
-            field_types = sheet_util.get_fields_types(self.settings.workbook_name,
-                                                      self.settings.sheet_name)
-            self.__validate_field_names_and_types_count(field_names, field_types)
-            self.__validate_field_names(field_names)
-        else:
-            for field in self.settings.fields:
-                field_name_type = field.split(':')
-                field_name = field_name_type[0]
-                field_names.append(field_name)
-                field_types.append(field_name_type[1])
-                self.__validate_field_names([field_name])
+        s3_key_with_partition = self.settings.target_s3_dir
```

Comment:
Put this logic in method `calculate_s3_key`

Reply:
Fixed